src/sql/exceptions.py

```diff
@@ -1,13 +1,18 @@
 from IPython.core import error
 
 
-def ArgumentError(message):
-    exc = error.UsageError(message)
-    exc.error_type = "ArgumentError"
-    return exc
+def exception_factory(name):
+    def _error(message):
+        exc = error.UsageError(message)
+        exc.error_type = name
+        return exc
 
+    return _error
 
-def TableNotFoundError(message):
-    exc = error.UsageError(message)
-    exc.error_type = "TableNotFoundError"
-    return exc
+
+ArgumentError = exception_factory("ArgumentError")
```

Comment:
Can we create a small guide that explains which error to use and when?

Reply:
I added a new section to the developer guide. Also added some new comments to this file explaining when to use each